### DISCUSSION TITLE
API features thread

### DISCUSSION THREAD

**Discussion by sunstep** [Freika/dawarich#178] [Ideas]
I wrote a thread earlier requesting some API changes. I need more changes to be made, so I decided to make a new thread that will contain all the requested changes. Of course, if you, the reader, need a certain thing from the API, and it is currently not implemented or suggested in this thread, feel free to comment and suggest.

### **Stats**
✅Retrieve total, yearly and monthly stats from API
* Retrieve daily stats from API
* Update stats using API

### **Points**
✅Add a query parameter to set how many points should be returned by the API and pagination to avoid timeout errors. 
✅Add total pages and current page to response headers.
✅ Option to reduce the data of each point to reduce the size of the output given by the API.
✅ Option to reverse returned results (sort by new instead of old)
* Add option to bulk delete points.
* Add a new endpoint so you are able to send (batch of) points to Dawarich without the endpoint being very specific to overland/ owntracks.
* Add a /points/delete/{id} endpoint and use /points/{id} for showing detailed information for a certain point (so retrieving a specific point from the API). It's not really a must-have change, but it's more logical in my opinion.

### **User management**
* Add endpoint to retrieve user information ‼️
* Add endpoint to update user information
✅ Add endpoint to modify Dawarich settings through API.
✅ Retrieve user settings from API.
✅ Modify user settings through API.

### **Import/ export management**
* Add option to initiate import
* Add option to initiate export for selected points
* Add option to destroy imports/ exports
* Add option to download exports.

### **Technical**
* Add option to manage sidekiq operations
* Add option to retrieve sidekiq operation information

These are just ideas, I think they would be nice to have the above features. I am not expecting anything, especially not that everything is provided through one update. The features can be added over time.

Edited the thread to make the requested changes more concrete.

**Comment by Freika** [comment]
## ✅ **Stats**

https://github.com/Freika/dawarich/releases/tag/0.10.0

**Reply by sunstep** [reply]
This is awesome, a good step forward for Dawarich API. I did forget to mention in this thread that we should also have an api endpoint for updating stats. I edited the post. Could this be implemented as well?

**Comment by Freika** [comment]
✅ ## Points API Pagination

https://github.com/Freika/dawarich/releases/tag/0.11.1

**Comment by sunstep** [comment]
## ✅Add total pages and current page to response headers.

https://github.com/Freika/dawarich/releases/tag/0.13.7

**Comment by mattisaoldgit** [comment]
Amazing Work/App! 

Not quite API topic, but could we get a webhook (that the user can config) to fire on a visit entry/exit? I am ok at python, but zero knowledge with Ruby, thats why I am asking instead of coding and offering. 

Once again great work

**Reply by Freika** [reply]
@mattisaoldgit it's requested and will be done at some point in the future

**Comment by sunstep** [comment]
# ✅ Option to reduce the data of each point to reduce the size of the output given by the API.

https://github.com/Freika/dawarich/releases/tag/0.14.4

**Comment by sunstep** [comment]
Bumping this thread so it goes back up in the discussion list.

**Comment by kobius77** [comment]
not sure if this is the kind of API functionality to be discussed here, but:

i started importing activities from strava (as they offer an API to do so), and had to find that having these points alongside the ones tracked by gpslogger at the same time can be suboptimal - at best. so, simple solution of course would be to turn off gpslogger while running, but then i'd likely forget to do so or turn it on again 9 out of 10 times.

so it would be cool if i could delete existing points in darawich within the timeframe of a strava activity with an API call before importing it. 🙆

**Comment by xathon** [comment]
I'd like to suggest a "current day/month" stats endpoint. For the year, it's easy, since you just get the first entry of the yearlyStats array, but for months that doesn't work, since all the months are there with "0" for months in the future.
